File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        
        if (s.length() % 2 == 1) {
            return 0;
        }
        
        for (int i=0; i<s.length(); i++) {
            Stack<Character> stack = new Stack<>();
            String subString = s.substring(i, s.length()) + s.substring(0, i);
            
            for (int j=0; j<subString.length(); j++) {
                char c = subString.charAt(j);
            
                if (stack.isEmpty()) {
                    stack.push(c);
                } else if (c == ')' && stack.peek() == '(') {
                    stack.pop();
                } else if (c == ']' && stack.peek() == '[') {
                    stack.pop();
                } else if (c == '}' && stack.peek() == '{') {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            }
        
            if (stack.isEmpty()) {
                answer += 1;
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    
    for i in range(len(s)):
        newList = s[i:] + s[:i]
        q = [newList[0]]
        for j in range(1, len(s)):
            if len(q) != 0:
                if q[-1] == '{' and newList[j] == '}' or q[-1] == '[' and newList[j] == ']' or q[-1] == '(' and newList[j] == ')':
                    q.pop()
                    continue
            q.append(newList[j])
                    
        if not q:
            answer += 1
    
    return answer